src/server/domain/consents/{ID}/generateChallenge.ts

```diff
@@ -0,0 +1,100 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+
+ --------------
+ ******/
+
+import { Request } from '@hapi/hapi'
+import { consentDB } from '../../../../lib/db'
+import { thirdPartyRequest } from '../../../../lib/requests'
+import { Consent } from '../../../../model/consent'
+import { Enum } from '@mojaloop/central-services-shared'
+import SDKStandardComponents from '@mojaloop/sdk-standard-components'
+
+/**
+ * Validates whether generate challenge request is valid
+ * by comparing consent ID sent matches with existing consent in table
+ * and if source ID matches initiator ID
+ * @param request: request received from PISP
+ * @param consent: Consent object
+ */
+export function isConsentRequestValid (
+  request: Request,
+  consent: Consent): boolean {
+  const fspiopSource = request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
+  return (consent && consent.initiatorId === fspiopSource)
+}
+
+/**
+ * Assigns credentials to given consent object and updates in the database
+ * Returns updated consent object
+ */
+export async function updateCredential (
```

Comment:
```suggestion
export async function updateConsentCredential (
```

Reply:
this code looks for me as a good candidate to be migrated to the ConsentDB model as a method `updateCredential(consent: Consent, credential: ConsentCredential)`, It would be wonderful if you could introduce a new ConsentCredential interface containing these three properties: challenge, status, type

Reply:
@eoln we actually tried many variations of `update` for `Consents` and decided on the correct one. The current one can take in any of the fields that you pass in and updates only those. This is because different endpoints update different parts of the `Consent` resource` and it wouldn't be feasible to have 3-4 variations of `update` for these things. Instead, we choose to stay usage agnostic in the model and let handler decide what to update.